### PR TITLE
NXBT-2500: Upgrade slave image to bionic

### PR DESCRIPTION
**Pull request by mguillaume**
Main changes:
pgfouine removed
easy_install removed -> no more funkload
python-pypdf replaced with python-pypdf2
postgresql 9.5 -> 10
mongodb 3.2 -> 4.0
ruby 2.3 -> 2.5
ccextractor 0.82 -> 0.87

### CODE REVIEW

File: roles/slave_tools/tasks/main.yml [hunk 1]
```diff
@@ -116,12 +116,14 @@
...
   tags: apt
+- name: Fix ImageMagick policy
+  replace: dest=/etc/ImageMagick-6/policy.xml regexp='policy domain="coder" rights="none" pattern="PDF"' replace='policy domain="coder" rights="read|write" pattern="PDF"'
 - apt: state=latest name=tcpwatch-httpproxy install_recommends=no
 - pip: name=logchart
 - pip: name=boto
 - pip: name=docker-py
 - pip: name=awscli
-- shell: easy_install -f http://funkload.nuxeo.org/snapshots/ -U funkload
+#- shell: easy_install -f http://funkload.nuxeo.org/snapshots/ -U funkload -> easy_install gone in bionic
```
Comment (jcarsique) [inline roles/slave_tools/tasks/main.yml]: remove the line

File: roles/slave_tools/tasks/main.yml [hunk 2]
```diff
@@ -94,8 +94,8 @@
   - python-docutils
   - gnuplot
   - python-reportlab
-  - python-pypdf
-  - pgfouine
+  - python-pypdf2
+  #- pgfouine -> pgfouine gone in bionic
```
Comment (jcarsique) [inline roles/slave_tools/tasks/main.yml]: remove the line

File: roles/slave_tools/tasks/main.yml [hunk 3]
```diff
@@ -116,12 +116,14 @@
   - libxml2-utils
   - libkrb5-dev
   tags: apt
+- name: Fix ImageMagick policy
+  replace: dest=/etc/ImageMagick-6/policy.xml regexp='policy domain="coder" rights="none" pattern="PDF"' replace='policy domain="coder" rights="read|write" pattern="PDF"'
```
Comment (jcarsique) [inline roles/slave_tools/tasks/main.yml]: check compliance with NXP-25914 and @efge 's sentence

File: roles/slave_tools/tasks/nodejs.yml [hunk 4]
```diff
@@ -19,6 +19,11 @@
   tags: nodejs
   environment:
     PATH: /opt/build/tools/node-v4.5.0-linux-x64/bin:{{ansible_env.PATH}}
+# Bypass unstable unicode.org
+- file: path=/usr/share/unicode-data state=directory
+  tags: nodejs
+- copy: src=UnicodeData.txt dest=/usr/share/unicode-data/UnicodeData.txt mode=0644 owner=root group=root
+  tags: nodejs
```
Comment (jcarsique) [inline roles/slave_tools/tasks/nodejs.yml]: Why?

File: roles/slave_tools/tasks/redis.yml [hunk 5]
```diff
@@ -7,4 +7,13 @@
...
   tags: redis
+- name: Do not listen on IPv6
+  lineinfile:
+    dest: /etc/redis/redis.conf
+    regexp: '::1'
+    state: absent
+- name: Listen on IPv4 loopback
+  lineinfile:
+    dest: /etc/redis/redis.conf
+    line: 'bind 127.0.0.1'
```
Comment (jcarsique) [inline roles/slave_tools/tasks/redis.yml]: Why?

File: roles/slave_tools/tasks/main.yml [hunk 6]
```diff
@@ -272,9 +274,9 @@
   npm: name=azure-cli version=0.10.7 global=yes
   register: azure_install
   tags: azure-cli
-- name: Add azure symlink
-  file: path=/usr/local/bin/azure src=/opt/build/tools/node-v6.9.2-linux-x64/bin/azure state=link
-  tags: azure-cli
+#- name: Add azure symlink
+#  file: path=/usr/local/bin/azure src=/opt/build/tools/node-v6.9.2-linux-x64/bin/azure state=link
+#  tags: azure-cli
```
[the remark this answers is not in the document]
Comment (mguillaume) [reply roles/slave_tools/tasks/main.yml]: Symlink is added by the package in this version

File: roles/slave_tools/tasks/nodejs.yml (hunk 4 above)
Comment (mguillaume) [reply roles/slave_tools/tasks/nodejs.yml]: unicode.org site is very unstable, often not responding. If the file is not available locally, it blocks the slave image build.

File: roles/slave_tools/tasks/redis.yml (hunk 5 above)
Comment (mguillaume) [reply roles/slave_tools/tasks/redis.yml]: It was the default before. In newer version, it also tries to bind on the IPv6 localhost, which is not available in containers, so we force the IPv4-only bind.

**Comment by jcarsique** [comment]
:+1:

**Comment by jcarsique** [comment]
:+1:

**Pull request merged**